### ISSUE TITLE
Allow customization of the temporary directory for `dvc exp --queue` 

### ISSUE THREAD

**Issue by abhi18av** [iterative/dvc#8980]
Hi team 👋 

Thanks for the wonderful tool!

I am exploring a use-case for DVC along with other tools which I rely upon in my projects and was wondering whether there's a way to customize the temporary directory for the experiments queue mechanism.

I have tried adding the `--temp` command but the results seems to be the same and each experiment still runs in a separate environment which downloads the large genomic files into each of those experiments.

```console
$ dvc exp run -S "annotation_tool=dfast,prokka" -S "outdir=results" --queue [[--temp]]
```

The feature request here is the (optional) ability to run the experiments in the same directory as opposed to individual directories (sensible default).

Happy to provide other details if needed.

**Comment by shcheklein** [comment]
@abhi18av 

> which downloads the large genomic files

Could you please give us a bit more details? Is it a custom script that downloads this data, is it DVC-tracked data, what kind of data / amount are we talking about?

> The feature request here is the (optional) ability to run the experiments in the same directory as opposed to individual directories (sensible default).

It means that you are running only one experiment at a time. Is it fine for your use case?

**Comment by abhi18av** [comment]
@shcheklein 

> Is it a custom script that downloads this data, is it DVC-tracked data

Yes, it downloads the genomic data from NCBI SRA archives and leads to datasets of hundreds of GBs - therefore I wouldn't want to track them via DVC. Maybe that's supported, but at the moment I'm still exploring the solution so please excuse my lack of familiarity or knowledge :) 

> It means that you are running only one experiment at a time. Is it fine for your use case?

Yup, that actually sounds ideal - I wouldn't want to run multiple experiments together and risk overwrites of the results.

**Comment by shcheklein** [comment]
Thanks @abhi18av ! Two workarounds that come to my mind:

- download data outside the repo (so that all experiments access the same data)
- Use DVC stage that outputs a downloaded data. If you setup it with one of the [links](https://dvc.org/doc/user-guide/data-management/large-dataset-optimization#file-link-types-for-the-dvc-cache) (besides `copy`) the way it's described [here](https://dvc.org/doc/user-guide/data-management/large-dataset-optimization#configuring-dvc-cache-file-link-type) + use `push: false` (as described [here](https://dvc.org/doc/user-guide/project-structure/dvcyaml-files#output-subfields)). This way you would avoid duplicates. The only overhead is creating links for each experiment. If there are dozens of thousands of files It might become slow to run hundreds of experiments.

Would one of these workaround work in your case?

**Comment by abhi18av** [comment]
Hey @shcheklein , 

I agree that these are the ideal solutions, however the use-case I have is a bit more invested. The scripts/pipelines can download the data during the runtime itself, depending on the parameters of the invocation.


But, I have a strong feeling that I need to experiment more with `dvc` before I could understand what's the best data tracking strategy which fits my use-case.

In the meantime, the experiment queue/location would be the short-term quick-win 🙏 😊
